src/main/kotlin/org/wycliffeassociates/otter/common/domain/resourcecontainer/ImportResourceContainer.kt

```diff
@@ -1,194 +1,156 @@
 
-    private fun constructContainerTree(rc: ResourceContainer): Tree {
-        val root = constructRoot(rc)
-        val categories = getCategories(rc)
-        val nodes = categories
-                .map { category ->
-                    val categoryTree = Tree(category)
-                    categoryTree.addAll(
-                            getProjectsInCategory(rc.manifest, category.slug)
-                                    .map {
-                                        constructProjectTree(rc.dir, it, "project")
-                                    }
-                    )
-                    return@map categoryTree
+                    return@flatMap collectionRepository
+                            .importResourceContainer(container, tree, container.manifest.dublinCore.language.identifier)
+                            .toSingle { Result.SUCCESS }
+                            .onErrorResumeNext(cleanUp(newDirectory, Result.IMPORT_ERROR))
                 }
-                .plus(
-                        getProjectsNotInCategories(rc.manifest, categories.map { it.slug })
-                                .map {
-                                    constructProjectTree(rc.dir, it, "project")
-                                }
-                )
-        root.addAll(nodes)
-        return root
+                .onErrorReturn { Result.UNKNOWN_ERROR }
+                .subscribeOn(Schedulers.io())
+
+    private fun cleanUp(containerDir: File, result: Result): Single<Result> = Single.fromCallable {
+        containerDir.deleteRecursively()
+        return@fromCallable result
     }
 
-    private fun constructProjectTree(containerDir: File, project: Project, type: String): Tree {
-        val projectDir = File(containerDir, project.path)
-        val files = projectDir.listFiles()
-        val book = Tree(Collection(project.sort, project.identifier, type, project.title, null))
-        val chapters = arrayListOf<Tree>()
-        for (file in files) {
-            val doc = ParseUsfm(file).parse()
-            for (chapter in doc.chapters) {
-                val slug = "${project.identifier}_${chapter.key}"
-                val col = Collection(chapter.key, slug, "chapter", chapter.key.toString(), null)
-                val tree = Tree(col)
-                // create a chunk for the whole chapter
-                val chapChunk = Content(0, "chapter",
-                        chapter.value.values.first().number,
-                        chapter.value.values.last().number, null)
-                tree.addChild(TreeNode(chapChunk))
-                for (verse in chapter.value.values) {
-                    val con = Content(verse.number, "verse", verse.number, verse.number, null)
-                    tree.addChild(TreeNode(con))
-                }
-                chapters.add(tree)
+    private fun validateResourceContainer(dir: File): Boolean = dir.listFiles().map { it.name }.contains("manifest.yaml")
```

Comment:
isn't this already handled by the resource container library?

Reply:
yes, but I think it's quicker to do the check here, especially since we just want to check that a manifest exists before copying to internal?